drivers/acpi/acpi.c

```diff
@@ -106,6 +107,11 @@ static struct RSDP_TABLE *lookup_for_rdsp(char *from)
 		if (!memcmp(p, RSDP_MAGIC, sizeof(RSDP_MAGIC)))
 			return (struct RSDP_TABLE *)p;
 
+	from = (char *)((*ebda_addr) << 4);
```

Comment:
Just curious why do we need to left shift 4 to content resides in *ebda_addr, it looks like segment in real mode. Since "the EBDA can be found in the two-byte location 40:0Eh on the BIOS data area."

Reply:
This is legacy way to scan RSDP with real mode. real mode 40:0Eh means 0x40e in physical address.
Please refer kernel implementation at drivers/acpi/acpica/tbxfroot.c:
acpi_status ACPI_INIT_FUNCTION
acpi_find_root_pointer(acpi_physical_address *table_address)
{
...
}

Reply:
LGTM

Reply:
As discussed afternoon, please add comment to indicate the reason why we need to left shift 4 bit of value which is read from EBDA.